main.go

```diff
@@ -181,8 +196,18 @@ func addLicense(path string, fmode os.FileMode, tmpl *template.Template, data *c
 	if err != nil {
 		return false, err
 	}
-	if hasLicense(b) || isGenerated(b) {
-		return false, err
+	if hasLicense(b) {
+		if isGenerated(b) || !updateOldLicense {
+			return false, nil
+		}
+		if isOutdated(b, data.Year) {
+			b, err := updateExistingLicense(b, data.Year)
+			if err != nil {
+				return false, err
+			}
+			return true, ioutil.WriteFile(path, b, fmode)
+		}
+		return false, nil
```

Comment:
What do you think about removing one of the `return false, nil` cases?
```suggestion
		if !isGenerated(b) && updateOldLicense && isOutdated(b, data.Year) {
			b, err := updateExistingLicense(b, data.Year)
			if err != nil {
				return false, err
			}
			return true, ioutil.WriteFile(path, b, fmode)
		}
		return false, nil
```

Reply:
Here was a bug, because for generated licences we shouldn't even check the license header.